src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,151 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            tail.next = new Node<>(tail, value, null);
+            tail = tail.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> nodeToInsert = new Node<>(null, value, null);
+        if (index == 0) {
+            nodeToInsert.next = head;
+            head.prev = nodeToInsert;
+            head = nodeToInsert;
+        } else {
+            Node<T> current = getNodeByIndex(index);
+            nodeToInsert.prev = current.prev;
+            nodeToInsert.next = current;
+            current.prev.next = nodeToInsert;
+            current.prev = nodeToInsert;
+        }
+        size++;
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        checkIndex(index);
+        Node<T> current = head;
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
```

Comment:
please be attentive to the common mistakes suggestions:
Don’t iterate your nodes from the head to tail or vice versa.
Pay attention to the index you receive as an input, you can iterate only first or second half of the list, depending on the index value. That will boost your lists’ performance.